presentation/src/main/java/com/example/pinterbest/presentation/viewmodels/ProfileViewModel.kt

```diff
@@ -45,6 +47,23 @@ class ProfileViewModel @Inject constructor(
     private val _checkAuthError = SingleLiveEvent<Int?>()
     val checkAuthError: LiveData<Int?> = _checkAuthError
 
+    fun getProfileDetailsById(userId: Int) {
+        viewModelScope.launch {
+            getProfileDetailsByIdUseCase(userId).collect { result ->
+                when (result) {
+                    is Result.Success -> {
+                        _profile.value = result.data
+                        _state.value = false
```

Comment:
I do not remember exactly, but please double check: probably you need to use `postValue` here instead of just `value = ...`

Reply:
No need